sensor_msgs/PointCloud2 input_cloud
----
sensor_msgs/PointCloud2[] cloud_array
geometry_msgs/Point[] centroid_array